test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Common.Linux
+{
+    using System;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Serilog;
+
+    public class PackageManagement
+    {
+        bool isInitialized = false;
+
+        string os;
+        string version;
+        string packageTool;
+        SupportedPackageExtension packageExtension;
+        string forceInstallConfigCmd;
+        string installCmd;
+        string iotedgeServices;
+        string uninstallCmd;
+
+        public enum SupportedPackageExtension
+        {
+            Deb,
+            Rpm
+        }
+
+        public PackageManagement()
+        {
+            this.os = null;
+            this.version = null;
+            this.packageTool = null;
+            this.packageExtension = 0;
+            this.forceInstallConfigCmd = null;
+            this.installCmd = null;
+            this.iotedgeServices = null;
+            this.uninstallCmd = null;
+            this.isInitialized = false;
+        }
+
+        async Task CheckInitializationAsync()
+        {
+            if (!this.isInitialized)
+            {
+                var tokenSource = new CancellationTokenSource();
+                string[] platformInfo = await Process.RunAsync("lsb_release", "-sir", tokenSource.Token);
+                this.InitializeParameters(platformInfo);
+                this.isInitialized = true;
+            }
+        }
+
+        void InitializeParameters(string[] platformInfo)
+        {
+            if (platformInfo.Length == 1)
```

Comment:
Is `platformInfo` sometimes not 1 line?

Reply:
nope, unfortunately. But they are returning in the same format though.